src/detect-ssh-software.c

```diff
@@ -50,8 +50,8 @@
 #define KEYWORD_NAME "ssh.software"
 #define KEYWORD_NAME_LEGACY "ssh_software"
 #define KEYWORD_DOC "ssh-keywords.html#ssh-software"
-#define BUFFER_NAME "ssh_software"
-#define BUFFER_DESC "ssh software"
+#define BUFFER_NAME "ssh.software"
```

Comment:
I'm a bit worried that this might break some config options. Each 'buffer name' is used to create a config name:
```
detect.mpm.<buffer_name>.shared
```
See https://github.com/OISF/suricata/blob/master/src/detect-engine-mpm.c#L236

Config wise, the following 2 are not equal
```
detect.mpm.ssh_software.shared
detect.mpm.ssh.software.shared
```
The latter would look like:
```yaml
detect:
  mpm:
    ssh:
      software:
        shared: true
```
The former:
```yaml
detect:
  mpm:
    ssh_software:
      shared: true
```

These config options are not part of the suricata.yaml.in, so perhaps nobody is using them. Maybe it would still be good to have a fallback option here:
```c
        /* see if we use a unique or shared mpm ctx for this type */
        int confshared = 0;
        char confstring[256] = "detect.mpm.";
        strlcat(confstring, n->name, sizeof(confstring));
        strlcat(confstring, ".shared", sizeof(confstring));
        if (ConfGetBool(confstring, &confshared) == 1)
            shared = confshared;
        else if (ConfGetBool(confstringwithunderscore, &confshared) == 1)
            shared = confshared;
```
Of course the ```confstringwithunderscore``` variable would have to be crafted somehow.